crates/hdx_parser/src/traits/declarations.rs

```diff
@@ -1,26 +1,31 @@
 use hdx_atom::{atom, Atom};
-use hdx_lexer::{Include, Kind, Token};
+use hdx_lexer::{Include, Kind};
 
-use crate::{discard, expect, expect_ignore_case, match_ignore_case, parser::Parser, unexpected, Result};
+use crate::{discard, expect, expect_ignore_case, parser::Parser, unexpected, Result};
 
 use super::Parse;
 
 pub trait Declaration<'a>: Sized + Parse<'a> {
 	type DeclarationValue: DeclarationValue<'a>;
 
 	fn parse_name(parser: &mut Parser<'a>) -> Result<Atom> {
-		match parser.next().clone() {
-			Token::Ident(atom) => {
+		let token = parser.next();
+		match token.kind() {
+			Kind::Ident => {
 				expect!(parser.next(), Kind::Colon);
-				Ok(atom.to_ascii_lowercase())
+				Ok(parser.parse_atom_lower(token))
```

Comment:
Is this the correct way to get a atom from a token ident now?

Reply:
Yep! The new Lexer avoids emitting or assigning Atoms until you explicitly call something like `parse_atom()`/`parse_atom_lower()`. This lets us skip a bunch of work+memory allocations.